commands/cli_nukeallresources_test.go

```diff
@@ -0,0 +1,547 @@
+	nukeResourcesShouldFail bool             // flag which decides whether NukeAllResources should return err
+	postNukeResources       []string         // what resourcetypes from test resources should exist after NukeAllResources
+}
+
+// getIAMRolePolicyMamp maps a role type to a corresponding policy
+func getTestRolePolicyMap() map[string]string {
+	return map[string]string{
+		"NoPerms": aws.TrimPolicyDocument(`{
+			"Version": "2012-10-17",
+			"Statement": {
+				"Effect": "Allow",
+				"Action": "ec2:DescribeRegions",
+				"Resource": "*"
+			}
+		}`),
+		"ROPerms": aws.TrimPolicyDocument(`{
+			"Version": "2012-10-17",
+			"Statement": {
+				"Effect": "Allow",
+				"Action": [
+					"ec2:Describe*",
+					"s3:List*",
+					"s3:Get*"
+				],
+				"Resource": "*"
+			}
+		}`),
+		"EC2ROS3RWPerms": aws.TrimPolicyDocument(`{
+			"Version": "2012-10-17",
+			"Statement": {
+				"Effect": "Allow",
+				"Action": [
+					"ec2:Describe*",
+					"s3:*"
+				],
+				"Resource": "*"
+			}
+		}`),
+	}
+}
+
+// TestNukeAllResourcesNoPerms tests deletion of test S3 and EC2 instance with an assumed
+// role which does not have S3/EC2 read/write permissions
+func TestNukeAllResources(t *testing.T) {
+	t.Parallel()
+	// Create a top level AWS session object which will be used to create/destroy roles
+	// and resources. Specifically use us-east-1 as we are running >= 8 tests and vCpu limits
+	// for eu-west-3 and some other regions are around - 8 - which allow only 4 instances to be created
+	awsParams, err := aws.NewCloudNukeAWSParams("us-east-1")
+	require.NoError(t, err, "Failed to setup AWS params")
+	testRolePolicyMap := getTestRolePolicyMap()
+
+	// Static assume role policy document which will allow the current identity to assume
+	// test roles
+	callerIdentityArn, err := aws.GetCallerIdentityArn(awsParams.AWSSession)
+	require.NoError(t, err, "Failed to get AWS caller identity")
+	assumeRolePolicyDocumentTmpl := aws.TrimPolicyDocument(`{
+		"Version": "2012-10-17",
+			"Statement": [
+			{
+				"Effect": "Allow",
+				"Principal": {
+					"AWS": "$callerIdentityArn"
+				},
+				"Action": "sts:AssumeRole"
+			}
+		]
+	}`)
+	assumeRolePolicyDocument := strings.Replace(
+		assumeRolePolicyDocumentTmpl, "$callerIdentityArn", callerIdentityArn, -1,
+	)
+
+	var testCases = []struct {
+		name                    string
+		createIAMRoleArgs       createIAMRoleArgs
+		createIAMRolePolicyArgs createIAMRolePolicyArgs
+		args                    testNukeAllResourcesArgs
+	}{
+		{
+			"AllPerms",
+			createIAMRoleArgs{
+				roleName: "", // Do not create any role
+			},
+			createIAMRolePolicyArgs{},
+			testNukeAllResourcesArgs{
+				getResourcesShouldFail:  false,
+				getResourcesOutputTypes: []string{"ec2", "s3"},
+				nukeResourcesShouldFail: false,
+				postNukeResources:       []string{},
+			},
+		},
+		{
+			"NoPerms",
+			createIAMRoleArgs{
+				roleName:                 "cloud-nuke-test-1-noperms",
```

Comment:
Does this name get used exactly as-is? If so, this test may fail due to a name conflict if we have two commits that kick off the test suite in parallel. Could you ensure that the role name has a unique ID appended to it using `util.UniqueId()`.

Reply:
Makes sense - will do.